app/controllers/api/v1/authentication_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Api::V1::AuthenticationController < ApplicationController
+    before_action :authorize_request, except: [:login, :create]
+    before_action :get_authorize_user, only: [:retrive]
+    before_action :set_user
+
+    def login
+        if @user and @user&.valid_password?(login_params[:password])
+            render json: { token: JsonWebToken.encode(user_id: @user.id) }, status: :ok
+        else
+            render json: { error: { email: ['El correo o la contraseña no son correctos'] }}, status: :unauthorized
+        end
+    end 
+
+    def create
+        if @user
+            render json: { errors: { email: ['El correo ya tiene una cuenta asociada'] }}
+        else
+            new_user = User.new(user_params)
+            new_user.add_role :admin
+            if new_user.save
+                render json: { token: token = JsonWebToken.encode(user_id: @user.id) }, status: :ok
+            else
+                render json: { errors: user.errors.messages }, status: :unprocessable_entity
+            end
+        end
+    end
+
+    def retrieve
+        render json: @current_user, status: :ok
+    end
+
+    private
+
+    def login_params
+        params.permit(:email, :password)
+    end
+
+    def user_params
+        params.permit(:first_name, :last_name, :email, :password, :password_confirmation, :phone)
+    end
+
+    def set_user
```

Comment:
What is the job of this method, i recommend to change the name to be more descriptive, something like "set_unautorized_user_by_email" that change completly the context and the meaning of the method, also it shouldn't be executed on all requests.

Reply:
In this case, can i use? 
```suggestion
    def get_admin_user_by_email
```

Reply:
yes it is better